github/github.go

```diff
@@ -739,7 +739,7 @@ func (c *Client) RateLimit() (*Rate, *Response, error) {
 		return nil, nil, err
 	}
 
-	return limits.Core, resp, err
+	return limits.Core, resp, nil
```

Comment:
Having high but practical test coverage is always a goal, and I think we're doing pretty well there (look at all the _test.go files). This finding doesn't change that. This is a particular rare edge case that I wouldn't expect to be fully caught by unit tests.

Also note that this is a deprecated method, so it's not worth spending a lot of time/effort on.